test/unit/aggregators/log-aggregator.test.js

```diff
@@ -175,6 +191,71 @@ test('Log Aggregator', async (t) => {
     logEventAggregator.addBatch(logs, priority)
     assert.equal(logEventAggregator.getEvents().length, 3)
   })
+
+  await t.test('add labels to logs when enabled', (t) => {
+    const { agent, commonAttrs, logEventAggregator, log } = t.nr
+    const expectedLabels = {
+      'tags.label1': 'value1',
+      'tags.LABEL2-ALSO': 'value3'
+    }
+    agent.config.loggingLabels = expectedLabels
+
+    logEventAggregator.add(log)
+    const payload = logEventAggregator._toPayloadSync()
+    assert.deepStrictEqual(payload, [
+      { common: { attributes: { ...commonAttrs, ...expectedLabels } }, logs: [log] }
+    ])
+  })
+
+  await t.test(
+    'should not add labels to logs when `application_logging.forwarding.enabled` is false',
+    (t) => {
+      const { agent, commonAttrs, logEventAggregator, log } = t.nr
+      const expectedLabels = {
+        'tags.label1': 'value1',
+        'tags.LABEL2-ALSO': 'value3'
+      }
+      agent.config.loggingLabels = expectedLabels
+      agent.config.application_logging.forwarding.labels.enabled = false
+
+      logEventAggregator.add(log)
+      const payload = logEventAggregator._toPayloadSync()
+      assert.deepStrictEqual(payload, [{ common: { attributes: { ...commonAttrs } }, logs: [log] }])
+    }
+  )
+
+  await t.test('should increment supportability metrics accordingly', (t) => {
+    const { agent } = t.nr
+    const expectedMetrics = [
+      [{ name: 'Supportability/Logging/Forwarding/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/LocalDecorating/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/Metrics/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/Labels/Nodejs/enabled' }]
+    ]
+    assertMetrics(agent.metrics, expectedMetrics, false, false)
```

Comment:
I think what you're really going for here is:

```suggestion
    assertMetrics(agent.metrics, expectedMetrics, { exact: false })
```